test/LocalVocabTest.cpp

```diff
@@ -7,36 +7,295 @@
+    // NOTE: The last argument is a string representation of the expression,
+    // which does not matter for the purposes of this test.
+    return {std::make_unique<sparqlExpression::GroupConcatExpression>(
+                false,
+                std::make_unique<sparqlExpression::VariableExpression>(
+                    Variable{variableName}),
+                separator),
+            "GROUP_CONCAT"};
+  };
+  GroupBy groupBy(
+      testQec, {Variable{"?x"}},
+      {Alias{groupConcatExpression("?y", "|"), Variable{"?concat"}}},
+      qet(values1));
+  checkLocalVocab(groupBy, std::vector<std::string>{"x", "y1", "y2", "y1|y2"});
+
+  // DISTINCT again, but after something has been added to the local vocabulary
+  // (to check that the "y1|y2" added by the GROUP BY does not also appear here,
+  // as it did before GroupBy::computeResult copied it's local vocabulary).
+  Distinct distinct2(testQec, qet(values1), {0});
+  checkLocalVocab(distinct2, std::vector<std::string>{"x", "y1", "y2"});
+
+  // UNION operation with exactly one non-empty local vocab and with two
+  // non-empy local vocabs.
+  Union union1(testQec, qet(values1), qet(values2));
+  checkLocalVocab(union1, std::vector<std::string>{"x", "y1", "y2"});
+  Union union2(testQec, qet(values1), qet(values1));
+  checkThrow(union2);
+
+  // MINUS operation with exactly one non-empty local vocab and with
+  // two non-empty local vocabs (the fourth argument are the indices of the
+  // matching columns).
+  Minus minus1(testQec, qet(values1), qet(values2), {});
+  checkLocalVocab(minus1, std::vector<std::string>{"x", "y1", "y2"});
+  Minus minus2(testQec, qet(values1), qet(values1), {});
+  checkThrow(minus2);
+
+  // FILTER operation (the third argument is an expression; which one doesn't
+  // matter for this test).
+  Filter filter(
+      testQec, qet(values1),
+      {std::make_unique<sparqlExpression::VariableExpression>(Variable{"?x"}),
+       "Expression ?x"});
+  checkLocalVocab(filter, std::vector<std::string>{"x", "y1", "y2"});
+
+  // BIND operation (the third argument is a `parsedQuery::Bind` object).
+  Bind bind(
+      testQec, qet(values1),
+      {{std::make_unique<sparqlExpression::VariableExpression>(Variable{"?x"}),
+        "Expression ?x"},
+       Variable{"?z"}});
+  checkLocalVocab(bind, std::vector<std::string>{"x", "y1", "y2"});
+
+  // TRANSITIVE PATH operation.
+  //
+  // NOTE: As part of an actual query, the child of a transitive path
+  // operation is always an index scan, which has a non-empty
+  // local-vocabulary. Still, it doesn't harm to test this.
+  TransitivePath transitivePath(testQec, qet(values1), true, true, 0, 1, {}, {},
+                                Variable{"?x"}, Variable{"?y"}, 1, 1);
+  checkLocalVocab(transitivePath, std::vector<std::string>{"x", "y1", "y2"});
+
+  // PATTERN TRICK operations.
+  //
+  // TODO: Requires an index with pattern. Maybe Johannes can tell me how to
+  // do this.
```

Comment:
It seems that we should simply enable the patterns in the test index from the helpers. I am a bit puzzled that those are not there. Are there also issues with the implementation of this propagations, because as far as I see, Those are still missing.